src/cdk/collections/tree-adapter.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {SelectionModel} from './selection';
+
+
+/**
+ * Tree Adapter to change any structured data type to flattened tree data.
+ * To be implemented by tree users.
+ */
+export interface TreeAdapter<T> {
```

Comment:
I'm wondering if a more specific name might be useful here. Maybe something like `TreeFlattener` or `FlatTreeAdapter`. Thoughts?

Reply:
`TreeDataNodeFlattener`?